bindings/java/Makefile

```diff
@@ -1,29 +1,89 @@
-.PHONY: gen_const clean jar all lib samples install
+all: jar lib samples
 
-all: gen_const
-	$(MAKE) -f Makefile.build all
+JC=javac
 
-lib:
-	$(MAKE) -f Makefile.build lib
+JAVA_HOME := $(shell readlink -f `which $(JC)` | sed "s:/bin/$(JC)::")
 
-samples:
-	$(MAKE) -f  Makefile.build samples
+JAVA_INC := $(shell realpath $(JAVA_HOME)/include)
 
-jar:
-	$(MAKE) -f Makefile.build jar
+JAVA_PLATFORM_INC := $(shell dirname `find $(JAVA_INC) -name jni_md.h`)
+
+UNICORN_INC=../../include
+
+SAMPLES := $(shell ls samples/*.java)
+TESTS := $(shell ls tests/*.java)
+SRC := $(shell ls unicorn/*.java)
+
+OS := $(shell uname)
+ifeq ($(OS),Darwin)
+   LIB_EXT=.dylib
+else ifeq ($(OS),Linux)
+   LIB_EXT=.so
+else
+   LIB_EXT=.dll
+endif
+
+CC=gcc
+CFLAGS=-fPIC
+LDFLAGS=-shared -fPIC
+LIBS=-lunicorn
+LIBDIR=-L../../
+INCS=-I$(JAVA_INC) -I$(JAVA_PLATFORM_INC) -I$(UNICORN_INC)
+
+CLASSPATH=./
+
+.SUFFIXES: .java .class
+
+tests/%.class: tests/%.java
+	$(JC) -Xlint:deprecation -classpath .:unicorn.jar:testdep/junit-4.13.2.jar $(JFLAGS) $<
+
+%.class: %.java
+	$(JC) -Xlint:deprecation -classpath .:unicorn.jar $(JFLAGS) $<
+
+OBJS=unicorn_Unicorn.o
+
+JARFILE=unicorn.jar
+
+%.o: %.c
+	$(CC) -c $(CFLAGS) $(INCS) $< -o $@
+
+unicorn_Unicorn.h: unicorn/Unicorn.java
+	$(JC) -h . $<
+
+unicorn_Unicorn.o: unicorn_Unicorn.c unicorn_Unicorn.h
+	$(CC) -O2 -Wall -Wextra -Wno-unused-parameter -c $(CFLAGS) $(INCS) $< -o $@
+
+libunicorn_java$(LIB_EXT): unicorn_Unicorn.o
+
+lib: libunicorn_java$(LIB_EXT) unicorn_Unicorn.h
+	$(CC) -o $< $(LDFLAGS) $(OBJS) $(LIBDIR) $(LIBS)
+
+samples: $(SAMPLES:.java=.class)
+tests: $(TESTS:.java=.class)
+jarfiles: $(SRC:.java=.class)
+
+jar: jarfiles
+	jar cf $(JARFILE) unicorn/*.class
+
+test: lib samples tests
+	java -Xcheck:jni -cp .:testdep/hamcrest-2.2.jar:testdep/junit-4.13.2.jar org.junit.runner.JUnitCore $(subst /,.,$(TESTS:.java=))
 
 install: lib jar
-	$(MAKE) -f Makefile.build install
+	cp libunicorn_java$(LIB_EXT) /usr/lib
+	cp $(JARFILE) /usr/share/java
 
 uninstall:
-	$(MAKE) -f Makefile.build uninstall
+	rm -f /usr/lib/libunicorn_java$(LIB_EXT)
+	rm -f /usr/share/java/$(JARFILE)
 
 gen_const:
-	cd .. && python3 const_generator.py java
+	cd .. && python const_generator.py java
```

Comment:
Why remove `python3` :)?

Reply:
Interesting, I hadn't even noticed. The old Makefile.build had `python`: https://github.com/unicorn-engine/unicorn/pull/1833/commits/93a5f9f3ef040b2f6ebedbfbf96c05e6e91c8ee7#diff-94ec5266cf434c0cb2f03d389e128624f0891519a56f84784479f13662f0e694L75.

Since `python3` seems to be universally used by the others, I'll fix this back.